meilisearch/src/search.rs

```diff
@@ -941,6 +1004,144 @@ pub fn perform_facet_search(
     })
 }
 
+pub fn perform_similar(
+    index: &Index,
+    query: SimilarQuery,
+    embedder_name: String,
+    embedder: Arc<Embedder>,
+) -> Result<SimilarResult, ResponseError> {
+    let before_search = Instant::now();
+    let rtxn = index.read_txn()?;
+
+    // using let-else rather than `?` so that the borrow checker identifies we're always returning here,
+    // preventing a use-after-move
+    let Some(internal_id) = index.external_documents_ids().get(&rtxn, &query.id)? else {
+        return Err(MeilisearchHttpError::DocumentNotFound(
+            query.id.into_inner(),
+            DocumentNotFoundRoute::Similar,
```

Comment:
Consider reverting DocumentNotFound to not take the route + use the ResponseError method below

Reply:
done